manager/redirect/utils.py

```diff
@@ -0,0 +1,2948 @@
+        3.436391113,
+        -2.158743,
+        FAType.SAA,
+        FASubnet.D,
+        12,
+        FAOrientation.NA,
+        None,
+        None,
+        152,
+        None,
+        [443, 439, 434, 435],
+        [443, 439, 434, 435, 436, 442, 433, 429, 438, 428, 430, 427],
+    ),
+    ForceActuatorData(
+        153,
+        441,
+        0.365734589,
+        4.00525,
+        -2.158743,
+        FAType.SAA,
+        FASubnet.A,
+        12,
+        FAOrientation.NA,
+        None,
+        None,
+        153,
+        None,
+        [442, 341, 337, 437, 438],
+        [437, 442, 341, 438, 337, 432, 443, 433, 338, 332, 439, 342],
+    ),
+    ForceActuatorData(
+        154,
+        442,
+        1.085088013,
+        3.872762939,
+        -2.158743,
+        FAType.SAA,
+        FASubnet.B,
+        10,
+        FAOrientation.NA,
+        None,
+        None,
+        154,
+        None,
+        [441, 437, 438, 439, 443],
+        [438, 443, 441, 439, 437, 433, 440, 432, 434, 337, 341, 427],
+    ),
+    ForceActuatorData(
+        155,
+        443,
+        1.60401001,
+        3.692779053,
+        -2.158743,
+        FAType.SAA,
+        FASubnet.C,
+        10,
+        FAOrientation.NA,
+        None,
+        None,
+        155,
+        None,
+        [442, 438, 439, 434, 440],
+        [439, 440, 442, 438, 434, 433, 435, 441, 428, 437, 432, 429],
+    ),
+]
+
+
+def force_actuator_from_id(actuator_id: int) -> ForceActuatorData:
+    """Returns ForceActuatorData with given actuator ID.
+
+    Parameters
+    ----------
+    actuator_id : `int`
+        Actuator ID - three digit number, with leading digit corresponding to
+        the force actuator index.
+
+    Returns
+    -------
+    force_actuator : ForceActuatorData
+        Record (ForceActuatorData class) for actuator with given actuator ID.
+
+    Raises
+    ------
+    RuntimeError
+        When actuator with given ID cannot be found.
+    """
+    # for fa in FATable:
+    #     print(fa.actuator_id)
+
+    try:
+        return next(fa for fa in FATable if fa.actuator_id == actuator_id)
+    except StopIteration:
+        raise RuntimeError(f"Cannot find force actuator with ID {actuator_id}")
+
+
+def actuator_id_to_index(
+    actuator_id: int, index_type: FAIndex = FAIndex.PRIMARY
+) -> int:
+    """Returns index for given actuator ID.
+
+    Parameters
+    ----------
+    actuator_id : `int`
+        Actuator ID - three digit number, with leading digit corresponding to
+        the force actuator index.
+    index_type : FAIndex
+        Index type. See FAIndex enumeration for possible values.
+
+    Returns
+    -------
+    index : `int`
+        Index_type index value of the force actuator with actuator ID ==
+        actuator_id.
+
+    Raises
+    ------
+    RuntimeError
+        When actuator with given ID cannot be found.
+    """
+    fa = force_actuator_from_id(actuator_id)
+    return fa.get_index(index_type)
+
+
+class CHRONOGRAF_SITES:
```

Comment:
Could that be turned into enum? Like:

class CHRONOGRAF_SITES(enum.IntEnum):
        SUMMIT = 1
        USDF = 2

etc..?


see https://github.com/lsst-ts/ts_xml/blob/develop/python/lsst/ts/xml/enums/MTM1M3.py for ane example.

Reply:
I'm a bit hesitant of doing this change. Having the class with a `str` hash gives me easier handling of the view as the `site` param is set as a `str` in the url (and then it is used on utils functions that needs the site param as a string), e.g:
```python
get_site_url_domain("summit")
``` 
Whereas `summit` can be `usdf`, `base` or `tucson` (added in case of future extensions). If I make the change to an enumeration, I'll need to create another hash class to make the conversion from the site in string to an enum.

Reply:
As discussed internally with @pkubanek , an StrEnum would be handy here, although we need to update to python > 3.11. A TODO was added for this.